chatsky/core/ctx_dict.py

```diff
@@ -0,0 +1,476 @@
+
+    @overload
+    async def __getitem__(self, key: int) -> BaseModel: ...  # noqa: E704
+
+    @overload
+    async def __getitem__(self, key: slice) -> List[BaseModel]: ...  # noqa: E704
+
+    async def __getitem__(self, key):
+        if isinstance(key, int) and key < 0:
+            key = self.keys()[key]
+        if self._storage is not None:
+            if isinstance(key, slice):
+                await self._load_items(
+                    [self.keys()[k] for k in range(len(self.keys()))[key] if k not in self._items.keys()]
+                )
+            elif key not in self._items.keys():
+                await self._load_items([key])
+        if isinstance(key, slice):
+            return [self._items[k] for k in self.keys()[key]]
+        else:
+            return self._items[key]
+
+    def __setitem__(self, key: Union[int, slice], value: Union[BaseModel, Sequence[BaseModel]]) -> None:
+        if isinstance(key, int) and key < 0:
+            key = self.keys()[key]
+        if isinstance(key, slice):
+            if isinstance(value, Sequence):
+                key_slice = self.keys()[key]
+                if len(key_slice) != len(value):
+                    raise ValueError("Slices must have the same length!")
+                for k, v in zip(key_slice, value):
+                    self[k] = v
+            else:
+                raise ValueError("Slice key must have sequence value!")
+        else:
+            self._keys.add(key)
+            self._added.add(key)
+            self._removed.discard(key)
+            self._items[key] = self._value_type.validate_python(value)
+
+    def __delitem__(self, key: Union[int, slice]) -> None:
+        if isinstance(key, int) and key < 0:
+            key = self.keys()[key]
+        if isinstance(key, slice):
+            for k in self.keys()[key]:
+                del self[k]
+        else:
+            self._removed.add(key)
+            self._added.discard(key)
+            self._keys.discard(key)
+            del self._items[key]
+
+    def __iter__(self) -> Sequence[int]:
+        return iter(self.keys() if self._storage is not None else self._items.keys())
+
+    def __len__(self) -> int:
+        return len(self.keys() if self._storage is not None else self._items.keys())
+
+    @overload
+    async def get(self, key: int, default=None) -> BaseModel: ...  # noqa: E704
+
+    @overload
+    async def get(self, key: Iterable[int], default=None) -> List[BaseModel]: ...  # noqa: E704
+
+    async def get(self, key, default=None):
+        """
+        Get one or many items from the dict.
+        Asynchronously load missing ones, if context storage is connected.
+        Raise an error if any requested elements are still missing after.
+
+        :param key: Key or slice for item retrieving.
+        :param default: Default value.
+        :return: One value or value list.
+        """
+
+        try:
+            return await self[key]
```

Comment:
This will always fail if `key` is `Iterable`.
This should be rewritten as
```python
await self.load_items(keys if isinstance(keys, Iterable) else [keys])
if isinstance(key, Iterable):
    return [self._items.get(k, default) for k in key]
else:
    return self._items.get(k, default)
```
Tests for using get with iterable key should be added.